torch_xla/core/dynamo_bridge.py

```diff
@@ -430,13 +430,22 @@ def extract_internal(xla_model: torch.fx.GraphModule):
         print(torch_xla._XLAC._get_xla_tensor_debug_info(xla_arg))
   # Don't reset the scope as we might be under some profiler trace scope.
   xm.mark_step(reset_scope=False)
+
+  # [Note: Dynamo real-time input-shape cache look-up]
+  # We maintain a mapping of input shapes to outputs of extract_graph_helper.
+  # When dynamic=True in torch.compile call, TorchDynamo will not trigger a
+  # new recompile. Then TorchXLA needs to figure out if these input shapes have
+  # been seen before.
+  input_shape_mappings = {}
```

Comment:
let's add a type hint here to make it more clear

Reply:
Added